File: frontend/src/services/boardService.js
import api from ".";

const boardService = {
  async getBoardList(params) {
    const { data } = await api.get("/board", { params });
    return data; // { list, count }
  },

  async getBoard(id) {
    const { data } = await api.get(`/board/${id}`);
    return data;
  },

  async createBoard(formData) {
    const { data } = await api.post("/board", formData, {
      headers: { "Content-Type": "multipart/form-data" },
    });
    return data;
  },

  async updateBoard(id, formData) {
    await api.put(`/board/${id}`, formData, {
      headers: { "Content-Type": "multipart/form-data" },
    });
  },

  async deleteBoard(id) {
    await api.delete(`/board/${id}`);
  },

  getFileUrl(fileName) {
    return `/api/board/files/${fileName}`;
  },

  /* 댓글 */
  async createReply(boardId, content) {
    const { data } = await api.post("/board/reply", { boardId, content });
    return data;
  },

  async updateReply(id, content) {
    await api.put(`/board/reply/${id}`, { content });
  },

  async deleteReply(id) {
    await api.delete(`/board/reply/${id}`);
  },
};

export default boardService;

File: backend/src/main/java/com/soaeng/happyhouse/board/service/BoardService.java
package com.soaeng.happyhouse.board.service;

import com.soaeng.happyhouse.board.dto.request.BoardParamDto;
import com.soaeng.happyhouse.board.dto.request.ReplyRequestDto;
import com.soaeng.happyhouse.board.dto.response.BoardDto;
import com.soaeng.happyhouse.board.dto.response.ReplyDto;
import org.springframework.web.multipart.MultipartHttpServletRequest;

import java.util.List;
import java.util.Map;

public interface BoardService {

    Long createBoard(MultipartHttpServletRequest request, String username);

    BoardDto getBoard(Long id, String username);

    void updateBoard(Long id, MultipartHttpServletRequest request);

    void deleteBoard(Long id);

    Map<String, Object> getBoardList(BoardParamDto param);

    ReplyDto createReply(ReplyRequestDto dto, String username);

    void updateReply(Long id, String content, String username);

    void deleteReply(Long id, String username);
}
